### ISSUE TITLE
Add ElasticSearch 7 params to old versions

### ISSUE THREAD

**Issue by PascalThesing** [elastic/elasticsearch-php#865]
### Summary of problem or feature request

I am using "elasticsearch/elasticsearch": 5.4.0 to support elasticsearch 5.0 and upper.
Since Elasticsearch 7.0 was released, type shouldn't be used for creating Mappings.
https://www.elastic.co/guide/en/elasticsearch/reference/master/removal-of-types.html

Since elasticsearch/elasticsearch put mapping requires a type for elasticsearch 5:
https://github.com/elastic/elasticsearch-php/blob/v5.4.0/src/Elasticsearch/Endpoints/Indices/Mapping/Put.php#L44
So, we can't drop it.

So it whould be awesome to include the param 
```
include_type_name
```
in the 
```
getParamWhitelist
```
Function of the Put.php.
https://github.com/elastic/elasticsearch-php/blob/v5.4.0/src/Elasticsearch/Endpoints/Indices/Mapping/Put.php#L63

and 
```
"rest_total_hits_as_int", "track_total_hits"
```
to the Search.php.

This whould enable us to be compatible to 5, 6 and 7.

### Code snippet of problem

Change 

```php
    /**
     * @return string[]
     */
    public function getParamWhitelist()
    {
        return array(
            'ignore_conflicts',
            'timeout',
            'master_timeout',
            'ignore_unavailable',
            'allow_no_indices',
            'expand_wildcards',
            'update_all_types'
        );
    }
```

to:

```php
    /**
     * @return string[]
     */
    public function getParamWhitelist()
    {
        return array(
            'ignore_conflicts',
            'timeout',
            'master_timeout',
            'ignore_unavailable',
            'allow_no_indices',
            'expand_wildcards',
            'update_all_types',
            'include_type_name'
        );
    }

```
And

```php
    public function getParamWhitelist()
    {
        return array(
            'analyzer',
            'analyze_wildcard',
            'default_operator',
            'df',
            'explain',
            'from',
            'ignore_unavailable',
            'allow_no_indices',
            'expand_wildcards',
            'indices_boost',
            'lenient',
            'lowercase_expanded_terms',
            'preference',
            'q',
            'query_cache',
            'request_cache',
            'routing',
            'scroll',
            'search_type',
            'size',
            'slice',
            'sort',
            'source',
            '_source',
            '_source_exclude',
            '_source_include',
            'stats',
            'suggest_field',
            'suggest_mode',
            'suggest_size',
            'suggest_text',
            'timeout',
            'version',
            'fielddata_fields',
            'docvalue_fields',
            'filter_path',
            'terminate_after',
            'stored_fields',
            'batched_reduce_size',
            'typed_keys'
        );
    }
```
```php
    public function getParamWhitelist()
    {
        return array(
            'analyzer',
            'analyze_wildcard',
            'default_operator',
            'df',
            'explain',
            'from',
            'ignore_unavailable',
            'allow_no_indices',
            'expand_wildcards',
            'indices_boost',
            'lenient',
            'lowercase_expanded_terms',
            'preference',
            'q',
            'query_cache',
            'request_cache',
            'routing',
            'scroll',
            'search_type',
            'size',
            'slice',
            'sort',
            'source',
            '_source',
            '_source_exclude',
            '_source_include',
            'stats',
            'suggest_field',
            'suggest_mode',
            'suggest_size',
            'suggest_text',
            'timeout',
            'version',
            'fielddata_fields',
            'docvalue_fields',
            'filter_path',
            'terminate_after',
            'stored_fields',
            'batched_reduce_size',
            'typed_keys',
            'rest_total_hits_as_int', 
            'track_total_hits'
        );
    }
```
-->

### System details

- Operating System
Linux/Windows/Max-Os
- PHP Version
7.2 or higher
- ES-PHP client version
5.4.0
- Elasticsearch version
5, 6 at the moment and hopefully 7

**Comment by ezimuel** [comment]
@PascalThesing I'm working to release elasticsearch-php 7.0.0 soon, I'm still fixing some issues for 6.7. Thanks for reporting this!

**Comment by ezimuel** [comment]
@PascalThesing I've already added `include_type_name` for `Elasticsearch\Endpoints\Indices\Mapping\Put` and  `rest_total_hits_as_int` in `Elasticsearch\Endpoints\Search` for 6.7.0. I'll add the missing `track_total_hits` for 7.0.0.

**Comment by PascalThesing** [comment]
Is there any reason for not adding track_total_hits to 6.7.0?

**Comment by ezimuel** [comment]
No reason, I'm considering to add it in 6.7.1.

**Comment by PascalThesing** [comment]
That would be awesome!

**Comment by ezimuel** [comment]
@PascalThesing, @shyim just released 6.7.1: https://github.com/elastic/elasticsearch-php/releases/tag/v6.7.1 with `track_total_hits` in search endpoint.
